core/Config/Cache.php

```diff
@@ -0,0 +1,89 @@
+<?php
+/**
+ * Piwik - free/libre analytics platform
+ *
+ * @link http://piwik.org
+ * @license http://www.gnu.org/licenses/gpl-3.0.html GPL v3 or later
+ */
+
+namespace Piwik\Config;
+
+use Piwik\Cache\Backend\File;
+use Piwik\Common;
+use Piwik\Filesystem;
+use Piwik\Piwik;
+use Piwik\Url;
+
+/**
+ * Exception thrown when the config file doesn't exist.
+ */
+class Cache extends File
+{
+    private $host = '';
+
+    public function __construct()
+    {
+        $this->host = $this->getHost();
+
+        // because the config is not yet loaded we cannot identify the instanceId...
+        // need to use the hostname
+        $dir = $this->makeCacheDir($this->host);
+
+        parent::__construct($dir);
+    }
+
+    private function makeCacheDir($host)
+    {
+        return PIWIK_INCLUDE_PATH . '/tmp/' . $host . '/cache/tracker';
```

Comment:
Should we check somewhere if this directory is writable?

Reply:
@diosmosis the directory might only be created when actually caching a file (the file backend does that). We already check if `PIWIK_INCLUDE_PATH . '/tmp/'` is writable as part of the system check not sure if it's enough?

Reply:
That sounds like enough 👍